columnflow/tasks/framework/mixins.py

```diff
@@ -1069,9 +1077,37 @@ class MLModelTrainingMixin(
     """
     A mixin class for training machine learning models.
     """
+    single_config = False
 
     @classmethod
-    def resolve_param_values(cls, params: dict[str, Any]) -> dict[str, Any]:
+    def build_taf_insts(cls, params: dict[str, Any], shifts: TaskShifts) -> dict[str, Any]:
+        # NOTE: we can only build TAF insts from the MLModel after ml_model_inst is set
+        if not cls.upstream_task_cls:
+            raise ValueError(f"upstream_task_cls must be set for multi-config task {cls.task_family}")
+
+        # run get_known_shifts for this task class
+        cls.get_known_shifts(params, shifts)
+
+        ml_model_inst = params["ml_model_inst"]
+        for config_inst, dataset_insts in ml_model_inst.used_datasets.items():
+            for dataset_inst in dataset_insts:
+                # NOTE: we need to copy here, because otherwise taf inits will only be triggered once
+                _params = params.copy()
+                _params["config_inst"] = config_inst
+                _params["config"] = config_inst.name
+                _params["dataset"] = dataset_inst.name
```

Comment:
Syntax sugar but also improves readability.

```suggestion
                _params = {
                    **params,
                    "config_inst": config_inst,
                    "config": config_inst.name,
                    "dataset": dataset_inst.name,
                }
```